ergotree-interpreter/src/eval/sglobal.rs

```diff
@@ -53,18 +54,115 @@ pub(crate) static XOR_EVAL_FN: EvalFn = |_mc, _env, _ctx, obj, args| {
+    }
+
+    let bytes_val = args
+        .first()
+        .cloned()
+        .ok_or_else(|| EvalError::NotFound("fromBigEndianBytes: missing bytes arg".to_string()))?;
+    let type_val = mc.tpe().t_range.clone();
+
+    let bytes = match bytes_val {
+        Value::Coll(CollKind::NativeColl(NativeColl::CollByte(bytes))) => bytes,
+        _ => {
+            return Err(EvalError::UnexpectedValue(format!(
+                "fromBigEndianBytes: expected first argument to be byte array, got {:?}",
+                bytes_val
+            )))
+        }
+    };
+
+    match *type_val {
+        SType::SByte => {
+            if bytes.len() != 1 {
+                return Err(EvalError::UnexpectedValue(
+                    "To deserialize Byte with fromBigEndianBytes, exactly one byte should be provided".to_string(),
+                ));
+            }
+            Ok(Value::Byte(bytes[0]))
+        }
+        SType::SShort => {
+            if bytes.len() != 2 {
+                return Err(EvalError::UnexpectedValue(
+                    "To deserialize Short with fromBigEndianBytes, exactly two bytes should be provided".to_string(),
+                ));
+            }
+            let b0 = bytes[0] as i16;
+            let b1 = bytes[1] as i16;
+            Ok(Value::Short(((b0 & 0xFF) << 8 | (b1 & 0xFF)) as i16))
+        }
+        SType::SInt => {
+            if bytes.len() != 4 {
+                return Err(EvalError::UnexpectedValue(
+                    "To deserialize Int with fromBigEndianBytes, exactly four bytes should be provided".to_string(),
+                ));
+            }
+            let bytes_vec: Vec<u8> = bytes.iter().map(|&x| x as u8).collect();
+            let int_bytes: [u8; 4] = bytes_vec.try_into().map_err(|_| {
+                EvalError::UnexpectedValue("Invalid byte array length for Int".to_string())
+            })?;
+            Ok(Value::Int(i32::from_be_bytes(int_bytes)))
+        }
+        SType::SLong => {
+            if bytes.len() != 8 {
+                return Err(EvalError::UnexpectedValue(
+                    "To deserialize Long with fromBigEndianBytes, exactly eight bytes should be provided".to_string(),
+                ));
+            }
+            let bytes_vec: Vec<u8> = bytes.iter().map(|&x| x as u8).collect();
+            let long_bytes: [u8; 8] = bytes_vec.try_into().map_err(|_| {
+                EvalError::UnexpectedValue("Invalid byte array length for Long".to_string())
+            })?;
+            Ok(Value::Long(i64::from_be_bytes(long_bytes)))
+        }
+        SType::SBigInt => {
+            if bytes.len() > 32 {
+                return Err(EvalError::UnexpectedValue(
+                    "BigInt value doesn't fit into 32 bytes in fromBigEndianBytes".to_string(),
+                ));
+            }
+            let bytes_vec: Vec<u8> = bytes.iter().map(|&x| x as u8).collect();
+            let big_int = num_bigint::BigInt::from_bytes_be(num_bigint::Sign::Plus, &bytes_vec);
```

Comment:
This would make it impossible to deserialize negative bigints, I suggest using BigInt256::from_be_slice instead

Reply:
Using BigInt256::from_be_slice now to handle negative bigints correctly.